tests/unittests/tests-libcose/tests-libcose.c

```diff
@@ -0,0 +1,215 @@
+﻿/*
+ * Copyright (C) 2018 Freie Universität Berlin
+ * Copyright (C) 2018 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup    tests
+ * @{
+ *
+ * @file
+ * @brief      Unit tests for pkg libcose
+ *
+ * @author     Koen Zandberg <koen@bergzand.net>
+ */
+
+#define MAX_NUMBER_BLOCKS 128
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "cn-cbor/cn-cbor.h"
+#include "cose.h"
+#include "cose/crypto.h"
+#include "embUnit.h"
+#include "memarray.h"
+#include "random.h"
+
+#include "tests-libcose.h"
+
+/* Example payload */
+static char payload[] = "Input string";
+/* Key ID's */
+static const char kid[] = "peter@riot-os.org";
+static const char kid2[] = "schmerzl@riot-os.org";
+/* Key pairs */
+static unsigned char pk[COSE_CRYPTO_SIGN_ED25519_PUBLICKEYBYTES];
+static unsigned char sk[COSE_CRYPTO_SIGN_ED25519_SECRETKEYBYTES];
+static unsigned char pk2[COSE_CRYPTO_SIGN_ED25519_PUBLICKEYBYTES];
+static unsigned char sk2[COSE_CRYPTO_SIGN_ED25519_SECRETKEYBYTES];
+static unsigned char symmkey[COSE_CRYPTO_AEAD_CHACHA20POLY1305_KEYBYTES];
+static uint8_t nonce[COSE_CRYPTO_AEAD_CHACHA20POLY1305_NONCEBYTES] = { 0 };
+/* COSE structs */
+static cose_sign_t sign, verify;
+static cose_key_t signer, signer2, symm;
+static cose_encrypt_t crypt, decrypt;
+/* COSE sign buffer */
+static uint8_t buf[1024];
+/*Signature Verification buffer */
+static uint8_t vbuf[1024];
+
+static cn_cbor block_storage_data[MAX_NUMBER_BLOCKS];
+static memarray_t storage;
+
+/* CN_CBOR calloc/free functions */
+static void *cose_calloc(size_t count, size_t size, void *memblock);
+static void cose_free(void *ptr, void *memblock);
+
+/* CN_CBOR block allocator context struct*/
+static cn_cbor_context ct =
+{
+    .calloc_func = cose_calloc,
+    .free_func = cose_free,
+    .context = &storage,
+};
+
+static void *cose_calloc(size_t count, size_t size, void *memblock)
+{
+    (void)count;
```

Comment:
You're not checking for correct allocations anymore?

Reply:
If with correct allocations you mean the counters to keep track of the number of allocated and freed blocks, then no. I removed them in favor of a simple and small test.

Reply:
Yes, I was asking for that. It's ok for me.